drawBot/drawBotDrawingTools.py

```diff
@@ -462,6 +463,38 @@ def pdfImage(self):
         self._drawInContext(context)
         return context.getNSPDFDocument()
 
+    def shareImage(self, format="pdf", service="airdrop", **kwargs):
+        """
+        Share the canvas to a service with a specified format.
+
+        As default the `format` is `pdf`, any suffix drawBot supports is possible.
+
+        `service` options are `airdrop`, `mail` or `message`.
+
+        .. downloadcode:: printImage.py
+
+            # set A4 page size
+            size(200, 200)
+            # draw something
+            text("Foo, bar", (10, 10))
+            # share it over airdrop
+            shareImage('pdf', service="airdrop")
+        """
+        path = tempfile.mkstemp(suffix=f".{format}")[1]
+
+        self.saveImage(path, **kwargs)
```

Comment:
Sounds like a job for `saveImage("NSImage")`. The tempfile just keeps hanging around.

Also: shouldn't this be a proper menu item rather than a library function?

Reply:
the cool thing to have this as library function is that it can be run from code outside the app.